### PR TITLE
Add OCI artifact platform support

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This allows pushing and pulling multiple platforms as OCI artifacts.

#### Which issue(s) this PR fixes:

Refers to #1482 
#### Does this PR have test?

Yes
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added support for platforms (`os[/arch][/variant][:os_version]`) when using seccomp OCI artifact profiles.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/security-profiles-operator-maintainers PTAL

File: installation-usage.md [hunk 1]
```diff
@@ -1986,6 +1989,114 @@ Please also note that signing is always required for push and pull. It is
 possible to add custom annotations to the security profile by using the
 `--annotations` / `-a` flag multiple times in `KEY:VALUE` format.
 
+### Using multiple platforms
+
+`spoc push` supports specifying the target platforms for the profiles to be
+pushed. This can be done by using the `--platforms` / `-p` together with the
+`--profiles` / `-p` flag. For example, to push two profiles into once artifact:
```
Comment (pjbgf) [inline installation-usage.md:1996]:
```suggestion
`--profiles` / `-p` flag. For example, to push two profiles into one artifact:
```

File: installation-usage.md [hunk 2]
```diff
@@ -1986,6 +1989,114 @@ Please also note that signing is always required for push and pull. It is
...
+    "org.opencontainers.image.created": "2023-04-28T08:59:17Z"
+  }
+}
+```
+
+There are a few fallback scenarios included in the CLI:
+
+- If neither a platform nor a input file is specified, then `spoc` will fallback
+  to the default profile (`/tmp/profile.yaml`) and platform
+  (`runtime.GOOS`/`runtime.GOARCH`)
```
Comment (pjbgf) [inline installation-usage.md:2058]:
```suggestion
  to the default profile (`/tmp/profile.yaml`) and platform
  (`runtime.GOOS`/`runtime.GOARCH`).
```

File: installation-usage.md [hunk 3]
```diff
@@ -1986,6 +1989,114 @@ Please also note that signing is always required for push and pull. It is
...
+- If only one platform is specified, then `spoc` will apply it and use the
+  default profile.
+- If only one input file is specified, then `spoc` will apply it and use the
+  default platform.
+- If multiple platforms and input files are provided, then `spoc` requires them
+  to match their occurrences. Platforms have to be unique as well.
+
+The Security Profiles Operator will try to pull the correct profile by using
+`runtime.GOOS`/`runtime.GOARCH`, but also falls back to the default profile
+(without any platform specified), if existing. `spoc pull` behaves in the same
```
Comment (pjbgf) [inline installation-usage.md:2068]:
```suggestion
(without any platform specified), if it exists. `spoc pull` behaves in the same
```

File: internal/pkg/artifact/artifact.go [hunk 4]
```diff
@@ -111,21 +116,31 @@ func (a *Artifact) Push(file, to, username, password string, annotations map[str
...
+	for platform, file := range files {
+		a.logger.Info(
+			"Adding profile " + file +
+				" for platform " +
+				platformToString(platform) +
+				" to store",
+		)
+		absPath, err := a.FilepathAbs(file)
+		if err != nil {
+			return fmt.Errorf("get absoluate file path: %w", err)
```
Comment (pjbgf) [inline internal/pkg/artifact/artifact.go:130]:
```suggestion
			return fmt.Errorf("get absolute file path: %w", err)
```

File: internal/pkg/cli/pusher/options.go [hunk 5]
```diff
@@ -53,11 +55,41 @@ func FromContext(ctx *ucli.Context) (*Options, error) {
...
+			if err != nil {
+				return nil, fmt.Errorf("parse platform %s: %w", platform, err)
+			}
+			parsedPlatforms = append(parsedPlatforms, parsedPlatform)
+		}
+
+		if len(profiles) == 0 {
+			options.inputFiles[parsedPlatforms[0]] = DefaultInputFile
+		} else if len(profiles) != len(platforms) {
+			return nil, errors.New("amount of profiles and platforms have to match")
```
Comment (pjbgf) [inline internal/pkg/cli/pusher/options.go:87]: Maybe?

```suggestion
			return nil, errors.New("number of profiles and platforms do not match")
```

**Review by pjbgf — COMMENTED** [review]
A few nits, otherwise lgtm.

**Comment by saschagrunert** [comment]
Thank you @pjbgf, I addressed all the comments.

**Pull request merged**